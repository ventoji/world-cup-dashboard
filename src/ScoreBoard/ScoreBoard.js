import React, { useState } from 'react';
import InputWC from '../InputWC';
import ButtonWC from '../ButtonWC';
import PropTypes from 'prop-types';

const SCORE_STYLE = {
  li: {
    display: 'flex',
    alignContent: 'space-around',
    marginBottom: '10px',
  },
};

const MatchStarted = ({ match, updateScore, handleFinish }) => {
  const [matchScore, setMatchScore] = useState({
    startAt: match.startAt,
    id: match.id,
    homeTeam: match.homeTeam,
    awayTeam: match.awayTeam,
    homeScore: 0,
    awayScore: 0,
  });

  const handleMatchScore = (event) => {
    // console.log(event.target.name);
    // console.log(event.target.value);

    setMatchScore({
      ...matchScore,
      [event.target.name]: event.target.value,
    });
  };

  const submitScore = (event) => {
    event.preventDefault();
    updateScore({
      id: matchScore.id,
      homeScore: matchScore.homeScore,
      awayScore: matchScore.awayScore,
    });
  };

  const submitFinishGame = (event) => {
    event.preventDefault();
    handleFinish(matchScore);
  };

  return (
    <li className="list-player" style={SCORE_STYLE.li}>
      <span style={{ color: 'blue' }}> Match: </span>{' '}
      <InputWC
        htmlId="homeScore"
        label={match.homeTeam}
        value={matchScore.homeScore}
        name="homeScore"
        type="number"
        style={{ width: '35%' }}
        onChange={handleMatchScore}
      />{' '}
      -
      <InputWC
        htmlId="awayScore"
        label={match.awayTeam}
        value={matchScore.awayScore}
        name="awayScore"
        type="number"
        style={{ width: '35%' }}
        onChange={handleMatchScore}
      />
      <ButtonWC
        label="update"
        style={{ height: '35%' }}
        onClick={submitScore}
      />
      <ButtonWC
        label="finish"
        style={{ height: '35%' }}
        onClick={submitFinishGame}
      />
    </li>
  );
};
const ScoreBoard = ({ games, updateScore, finishGame }) => {
  const shouldRender = Object.keys(games).length !== 0;

  const handleFinish = (matchScore) => {
    console.log(matchScore);
    finishGame(matchScore);
  };

  return shouldRender ? (
    <ul className="list-matches">
      {games.map((match, index) => (
        <MatchStarted
          key={match.id}
          match={match}
          updateScore={updateScore}
          handleFinish={handleFinish}
        />
      ))}
    </ul>
  ) : (
    <span> No match is playing now</span>
  );
};

InputWC.propTypes = {
  /** Array of active games */
  games: PropTypes.array,
};

ScoreBoard.defaultProps = {
  games: [],
};

export default ScoreBoard;
