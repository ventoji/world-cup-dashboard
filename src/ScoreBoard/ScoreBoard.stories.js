import React from 'react';
import ScoreBoard from './ScoreBoard';
import { withKnobs } from '@storybook/addon-knobs';

export default {
  title: 'WC Components/ScoreBoard',
  component: ScoreBoard,
  decorators: [withKnobs],
};

const startedGames = [
  {
    id: 1,
    startAt: 1616113389731,
    homeTeam: 'Mexico',
    awayTeam: 'Canada',
    homeScore: 0,
    awayScore: 0,
  },
  {
    id: 2,
    startAt: 1616113389731,
    homeTeam: 'Spain',
    awayTeam: 'Brasil',
    homeScore: 0,
    awayScore: 0,
  },
];

const Template = () => <ScoreBoard games={startedGames} />;

export const Default = Template.bind({});
