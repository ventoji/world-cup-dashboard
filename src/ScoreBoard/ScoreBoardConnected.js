import React from 'react';
import { connect } from 'react-redux';
import ScoreBoard from './ScoreBoard';
// import { listMatches } from '../utils';
import {
  updateMatchWC,
  finishMatchWC,
  showMatchResultWc,
} from '../store/actions/actions';

const ScoreBoardConnected = ({
  games,
  updateMatchWC,
  finishMatchWC,
  showMatchResultWc,
}) => {
  const updateScore = (matchScore) => {
    updateMatchWC(matchScore);
  };

  const finishGame = (matchScore) => {
    showMatchResultWc(matchScore);
    finishMatchWC(matchScore.id);
  };

  return (
    <ScoreBoard
      games={games}
      updateScore={updateScore}
      finishGame={finishGame}
    />
  );
};

const mapStatetoProps = (state) => {
  console.log(state);
  return {
    games: state.activeGames,
  };
};

const mapDispatchToProps = (dispatch) => ({
  updateMatchWC: (matchScore) => dispatch(updateMatchWC(matchScore)),
  finishMatchWC: (id) => dispatch(finishMatchWC(id)),
  showMatchResultWc: (matchScore) => dispatch(showMatchResultWc(matchScore)),
});

export default connect(
  mapStatetoProps,
  mapDispatchToProps,
)(ScoreBoardConnected);
