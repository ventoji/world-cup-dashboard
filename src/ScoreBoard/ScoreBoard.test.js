import React from 'react';
import ScoreBoard from './ScoreBoard';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('ScoreBoard  Component', () => {
  let scoreBoard;
  const games = [
    {
      id: 1,
      startAt: 1616113389731,
      homeTeam: 'Mexico',
      awayTeam: 'Canada',
      homeScore: 0,
      awayScore: 0,
    },
    {
      id: 2,
      startAt: 1616113389731,
      homeTeam: 'Spain',
      awayTeam: 'Brasil',
      homeScore: 0,
      awayScore: 0,
    },
  ];

  beforeEach(() => {
    scoreBoard = shallow(
      <ScoreBoard games={games} updateScore={() => {}} finishGame={() => {}} />,
    );
  });

  it('Should be defined', () => {
    expect(ScoreBoard).toBeDefined();
  });

  it('Should render properly', () => {
    expect(toJson(scoreBoard)).toMatchSnapshot();
    let element = scoreBoard.find('MatchStarted');
    expect(element).toHaveLength(2);
  });
});
