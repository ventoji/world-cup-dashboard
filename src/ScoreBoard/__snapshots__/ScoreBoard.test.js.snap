// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreBoard  Component Should render properly 1`] = `
<ul
  className="list-matches"
>
  <MatchStarted
    finishGame={[Function]}
    handleFinish={[Function]}
    key="1"
    match={
      Object {
        "awayScore": 0,
        "awayTeam": "Canada",
        "homeScore": 0,
        "homeTeam": "Mexico",
        "id": 1,
        "startAt": 1616113389731,
      }
    }
    updateScore={[Function]}
  />
  <MatchStarted
    finishGame={[Function]}
    handleFinish={[Function]}
    key="2"
    match={
      Object {
        "awayScore": 0,
        "awayTeam": "Brasil",
        "homeScore": 0,
        "homeTeam": "Spain",
        "id": 2,
        "startAt": 1616113389731,
      }
    }
    updateScore={[Function]}
  />
</ul>
`;
